Source/Scene/Primitive.js

```diff
@@ -683,6 +683,30 @@ define([
         return renamedVS + '\n' + showMain;
     };
 
+    Primitive._appendDistanceDisplayConditionToShader = function(primitive, vertexShaderSource) {
+        if (!defined(primitive._attributeLocations.distanceDisplayCondition)) {
+            return vertexShaderSource;
+        }
+
+        var renamedVS = ShaderSource.replaceMain(vertexShaderSource, 'czm_non_distanceDisplayCondition_main');
+        var distanceDisplayConditionMain =
+            'attribute vec3 boundingSphereCenter3DHigh; \n' +
+            'attribute vec3 boundingSphereCenter3DLow; \n' +
+            'attribute float boundingSphereRadius; \n' +
+            'attribute vec2 distanceDisplayCondition; \n' +
+            'void main() \n' +
+            '{ \n' +
+            '    czm_non_distanceDisplayCondition_main(); \n' +
+            '    vec4 centerRTE = czm_computeBoundingSphereCenter(); \n' +
```

Comment:
Nope, its generated just like `czm_computePosition` for all other geometries and the functions for getting adjacent points for polylines.